### PR TITLE
Improve invalid markup documentation section

### LINKED ISSUE #77: Broken doctype without space
A single space within doctype should be preserved.

Example:
```python
>>> from minify_html import minify
>>> minify('<!doctype html>')
'<!doctypehtml>'
```
Expected: `<!doctype html>`

### PR DESCRIPTION
**Pull request by adamchainz**
Fixes #77. I copied the suggestion from @roskakori and edited it a little.

### CODE REVIEW

**Comment by roskakori** [comment]
Thanks!

File: README.md [hunk 1]
```diff
@@ -190,7 +190,14 @@ All [`Cfg` fields](https://docs.rs/minify-html/latest/minify_html/struct.Cfg.htm
...
 
-Note that some of the minification done can result in HTML that will not pass validation, but remain interpreted and rendered correctly by the browser; essentially, the laxness of the browser is taken advantage of for better minification. These can be turned off via the `Cfg` object.
+Note that some of the minification done can result in HTML that will not pass validation, but remain interpreted and rendered correctly by browsers.
+Essentially, browser laxness is taken advantage of, for better minification.
+
+To produce valid HTML, set the following options in the `Cfg` object:
+
+    do_not_minify_doctype=True
+    ensure_spec_compliant_unquoted_attribute_values=True
+    keep_spaces_between_attributes=True
```
Comment (Rongronggg9) [inline README.md:200]:
```suggestion
To retain the validity of HTML, set the following options in the `Cfg` object:

    do_not_minify_doctype=True
    ensure_spec_compliant_unquoted_attribute_values=True
    keep_spaces_between_attributes=True

Note: minify-html does not aim to fix invalid HTML. Inputting an invalid HTML, even with the above options set, could sometimes not result in a valid HTML.
```

e.g. minify-html would not add `<!DOCTYPE html>` if it does not exist, would not upper-case `doctype` but only retain its case.

**Review by Rongronggg9 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (wilsonzlin) [reply README.md:200]: Yes this is a good point, if the input is invalid, the output will most likely remain so.

**Comment by wilsonzlin** [comment]
I've updated the README to mention the options to set in order to keep the output spec-compilant.

**Pull request closed without merging**